libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -91,13 +91,16 @@ string IRGenerator::generate(ContractDefinition const& _contract)
 					<dispatch>
 					<runtimeFunctions>
 				}
+				<dependanciesSource>
```

Comment:
We should make the dependencies more fine-grained. The runtime and the creation objects may not need the same set of contracts. Is it like that in the old code generator? In principle, the creation context could also access data in the runtime object.

Reply:
Hm, I'll check old codegen. But I wasn't able to compile source until I added dependencies to both of runtime and creation objects. Both of them are requesting `datasize(creation_object)` because of call to new